File: public/examples/integrations/mcp-servers/github_api/create_global_webhook_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "GithubApi.CreateGlobalWebhook";

// Start the authorization process
const authResponse = await client.tools.authorize({
  tool_name: TOOL_NAME,
  user_id: USER_ID,
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "payload_delivery_url": "https://example.com/webhook",
  "webhook_type": "web",
  "hmac_key_for_signature": "my_secret_key",
  "payload_content_type": "json",
  "send_notifications": true,
  "ssl_verification": "0",
  "trigger_events": [
    "push",
    "pull_request"
  ]
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

File: public/examples/integrations/mcp-servers/github_api/create_global_webhook_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "GithubApi.CreateGlobalWebhook"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'payload_delivery_url': 'https://example.com/webhook',
    'webhook_type': 'web',
    'hmac_key_for_signature': 'my_secret_key',
    'payload_content_type': 'json',
    'send_notifications': True,
    'ssl_verification': '0',
    'trigger_events': ['push', 'pull_request']
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))
